core/src/services/s3/core.rs

```diff
@@ -275,6 +275,7 @@ impl S3Core {
                     .for_each(|b| crc = crc32c::crc32c_append(crc, &b));
                 Some(BASE64_STANDARD.encode(crc.to_be_bytes()))
             }
+            Some(ChecksumAlgorithm::Md5) => Some(format_content_md5(body.to_bytes().as_ref())),
```

Comment:
Do we have a better way to handle this, like calculating in a stream instead of calling `body.to_bytes()` first? Our users might send data up to 5GiB in a single request.

Reply:
:+1: made a non-copy version `format_content_md5_iter` that works on iterator